fe/fe-core/src/test/java/com/starrocks/sql/plan/PlanFragmentTest.java

```diff
@@ -5554,4 +5559,18 @@ public void testJoinOutput() throws Exception {
                 "  |  cardinality: 1"));
         Assert.assertTrue(plan.contains("output columns: 2, 6"));
     }
+
+    @Test
+    public void testSemiJoinReorderWithProject() throws Exception {
```

Comment:
JoinAssociativityRule may have the same problem

Reply:
I was fixed it by other ways